### PR TITLE
Add challenge view all submissions page

### PR DESCRIPTION
**Pull request by Sanji515**
#### Changes proposed in this pull request:

- Created new component for view all submissions page under challenge component

- For selecting the phase using select box used the reusable component `selectPhaseComponent`

- Link to live demo: http://pr-152-evalai.surge.sh 

**<p align="center">Screenshot 1:</p>**

Screenshot from 2019-06-19 14-18-43

**<p align="center">Screenshot 2:</p>**

Screenshot from 2019-06-19 14-20-18

**<p align="center">Screenshot 3:</p>**

Screenshot from 2019-06-19 14-20-35

### CODE REVIEW

**Comment by galipremsagar** [comment]
@Sanji515, resolve conflicts of this now that dependent PR's are merged?

**Comment by Sanji515** [comment]
@galipremsagar @lunayach @RishabhJain2018 @Shekharrajak 

Resolved the conflicts, please review.

**Review by galipremsagar — COMMENTED** [review]
Overall looks good, have few comments will be taking them offline

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -21,6 +21,9 @@ import { ChallengeparticipateComponent } from './components/challenge/challengep
 import { ChallengeleaderboardComponent } from './components/challenge/challengeleaderboard/challengeleaderboard.component';
 import { ChallengesubmitComponent } from './components/challenge/challengesubmit/challengesubmit.component';
 import { ChallengesubmissionsComponent } from './components/challenge/challengesubmissions/challengesubmissions.component';
+import {
+  ChallengeviewallsubmissionsComponent
+} from './components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component';
```
Comment (lunayach) [inline src/app/app-routing.module.ts]: This could be in one line.

**Review by galipremsagar — APPROVED** [review]
LGTM, this can go in @RishabhJain2018

File: src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+
+  /**
+   * Is phase selected
+   */
+  isPhaseSelected = false;
+
+  /**
+   * Highlighted submission
+   */
+  submissionHighlighted: any = null;
```
Comment (lunayach) [inline src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts]: Are we using this?

File: src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+      } else if (this.fileSelected === '') {
+        this.globalService.showToast('error', 'The file type requested is not valid!');
+      }
+    }
+  }
+
+  /**
+   * load data with pagination
+   */
+  loadPaginationData = function(url) {
```
Comment (lunayach) [inline src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts]:
```suggestion
  loadPaginationData(url) {
```

File: src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+    if (url !== null) {
+      const SELF = this;
+      const API_PATH = url.split('localhost:8000/api/')[1];
```
Comment (lunayach) [inline src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts]: Get this `localhost:8000/api/` from environment file.

**Review by lunayach — COMMENTED** [review]
Functionality-wise LGTM. Just a few minor changes are required.

File: src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts (hunk 4 above)
Comment (Sanji515) [reply src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts]: @lunayach  done

File: src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts (hunk 3 above)
Comment (Sanji515) [reply src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts]: done

File: src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts (hunk 2 above)
Comment (Sanji515) [reply src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts]: no, removed it

**Review by lunayach — APPROVED** [review]

File: src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.scss [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+.cbx-label {
+  margin-left: 55px;
+  margin-right: 55px;
+}
+
+.cbx:checked ~ .cbx-label {
+  background: #4CAF50;
+}
```
Comment (RishabhJain2018) [inline src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.scss]: Please add a blank line here.

File: src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+          <button [ngClass]="paginationDetails.isNext" class="btn-floating
+            btn-pagination waves-effect waves-light"
+            (click)="loadPaginationData(paginationDetails.next)">
+            <i class="fa fa-chevron-right"></i>
+          </button>
+        </div>
+      </div>
+    </div>
+  </div>
+</div>
```
Comment (RishabhJain2018) [inline src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html]: Please add a blank line here.

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]

File: src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.scss (hunk 5 above)
Comment (Sanji515) [reply src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.scss]: done

File: src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html (hunk 6 above)
Comment (Sanji515) [reply src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.html]: done

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**